File: icli/lib/utils/ui.rb
# frozen_string_literal: true

module ICLI
  # UI handles prints and visual feedbacks
  class UI
    def echo(msg, color = nil)
      puts colorize(msg, color)
    end

    def error(msg)
      echo msg, 'red'
      exit(1)
    end

    private

    def colorize(str, color)
      case color
      when 'red'
        "\e[31m#{str}\e[0m"
      when 'green'
        "\e[32m#{str}\e[0m"
      when 'yellow'
        "\e[33m#{str}\e[0m"
      when 'blue'
        "\e[34m#{str}\e[0m"
      when 'cyan'
        "\e[36m#{str}\e[0m"
      else
        str
      end
    end
  end
end

File: lib/utils/ui.dart
import 'dart:io';

class UI {
  error(String msg) {
    echo(msg, Color.red);
    exit(1);
  }

  echo(String? text, Color color) {
    String? msg = "";

    switch (color) {
      case Color.red:
        msg = _addColor(text, 31);
        break;
      case Color.green:
        msg = _addColor(text, 32);
        break;
      case Color.yellow:
        msg = _addColor(text, 33);
        break;
      case Color.blue:
        msg = _addColor(text, 34);
        break;
      case Color.cyan:
        msg = _addColor(text, 36);
        break;
      default:
        msg = text;
    }

    print(msg);
  }

  String _addColor(String? msg, int code) {
    final stringCode = code.toString();
    final section = [stringCode, "m", msg].join();
    return "\x1B[$section\x1B[0m";
  }
}

enum Color { red, green, blue, cyan, yellow }
